### PR TITLE
Hokusai/Docker updates

### PR DESCRIPTION
**Pull request by eessex**
Addresses https://artsyproduct.atlassian.net/browse/PLATFORM-1245

In Hokusai config files have been run through prettier and named ports, `readinessProbe`, autoscaling replicas, and `resources` block have been added.

In Dockerfile, app is now started from a `deploy` user and uses `dumb-init` for entrypoint.

A separate ticket has been added to address running the app with SSL and updating dns to use `artsy.net`: https://artsyproduct.atlassian.net/browse/PLATFORM-1782

### CODE REVIEW

**Comment by eessex** [comment]
Looks like tests were not run 🤔

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
 # Cache elixir deps
 ADD mix.exs mix.lock ./
-RUN mix do deps.get, deps.compile
 
-RUN mix compile
+# Install app dependencies
+RUN mix do deps.get, deps.compile && \
+    mix compile
 
-ENV PORT 4000
```
Comment (eessex) [inline Dockerfile]: Removed this optimistically after adding to the hokusai config file

File: hokusai/production.yml [hunk 2]
```diff
@@ -21,13 +20,33 @@ spec:
...
           image: 585031190124.dkr.ecr.us-east-1.amazonaws.com/aprb:production
           imagePullPolicy: Always
           ports:
-          - containerPort: 4000
+            - name: aprb-http
+              containerPort: 4000
+          readinessProbe:
+            httpGet:
+              port: aprb-http
+              path: /ping
```
Comment (eessex) [inline hokusai/production.yml]: This may actually be `/api/ping` @ashkan18 can you confirm?

Comment (ashkan18) [reply hokusai/production.yml]: no, `ping` is correct 🙌 currently http://aprb.artsy.systems/ping

File: hokusai/production.yml [hunk 3]
```diff
@@ -21,13 +20,33 @@ spec:
...
+              httpHeaders:
+                - name: X-FORWARDED-PROTO
+                  value: https
+            initialDelaySeconds: 5
+            periodSeconds: 15
+            timeoutSeconds: 10
+          resources:
```
Comment (ashkan18) [inline hokusai/production.yml]: do we have to set these limits? my guess is this app uses very limited CPU and memory, maybe we only need to set some low limits?

File: hokusai/staging.yml [hunk 4]
```diff
@@ -21,28 +20,64 @@ spec:
...
+kind: HorizontalPodAutoscaler
+metadata:
+  name: aprb-web
+  namespace: default
+spec:
+  scaleTargetRef:
+    apiVersion: extensions/v1beta1
+    kind: Deployment
+    name: aprb-web
+  minReplicas: 2
```
Comment (ashkan18) [inline hokusai/staging.yml]: i think this can be 1 and max replicas maybe 2?

Comment (eessex) [reply hokusai/staging.yml]: We've been using a minimum of 2 and then `minReplicas` x3 as the max for these, but in staging could probably get by with 3 as the max.

File: hokusai/production.yml (hunk 3 above)
Comment (eessex) [reply hokusai/production.yml]: These should be set on everything, but potentially these could be lower -- will check out some other smaller apps to see how low they get, maybe @izakp has ideas on a better minimum baseline.

**Comment by eessex** [comment]
@ashkan18 so the issue of PR builds not running in CI pre-dates the recent changes to the Circle setup?

**Comment by ashkan18** [comment]
yes! its looks strangely random to me, for example https://github.com/artsy/aprb/pull/170 didn't run the tests while https://github.com/artsy/aprb/pull/174 one did!

**Comment by eessex** [comment]
Just toggled this setting on - maybe the culprit?
Screen Shot 2019-09-18 at 9 23 43 AM

**Comment by ashkan18** [comment]
yes, maybe! 🙌

**Comment by ashkan18** [comment]
oh picked it up now!!

File: hokusai/production.yml (hunk 3 above)
Comment (eessex) [reply hokusai/production.yml]: Bumped these down significantly

**Comment by ashkan18** [comment]
thanks! looks great to me! feel free to merge when ready. Once its merged, we can try staging and make sure things still work by checking `#apr-staging` slack channel and use `/apr-staging subscriptions` command in there to make sure it replies to slack commands.

**Pull request merged**